gateway/requests/requests.go

```diff
@@ -90,3 +90,8 @@ type AsyncReport struct {
 type DeleteFunctionRequest struct {
 	FunctionName string `json:"functionName"`
 }
+
+// SecretRequest create, update a secret
+type SecretRequest struct {
```

Comment:
How will the map work? CLUD using HTTP verbs?

Reply:
Right - It should be along the lines of secretName secretValue, then L part of CLUD will return a list of this type minus the value

Reply:
@ewilde what are your thoughts? It I have the impress that it may be slightly orthogonal to the existing design.

Reply:
I've made some commits since the start of this thread. I reverted my last one since I think we want to stick with one secret/request (POST,PUT,DELETE) approach.